src/draw/nxp/vglite/lv_draw_vglite.c

```diff
@@ -282,9 +288,26 @@ static int32_t _vglite_dispatch(lv_draw_unit_t * draw_unit, lv_layer_t * layer)
     /* Try to get an ready to draw. */
     lv_draw_task_t * t = lv_draw_get_next_available_task(layer, NULL, DRAW_UNIT_ID_VGLITE);
 
-    if(t == NULL || t->preferred_draw_unit_id != DRAW_UNIT_ID_VGLITE)
+    if(t == NULL)
         return LV_DRAW_UNIT_IDLE;
 
+    if(lv_draw_get_unit_count() > 1) {
+        /* Let the SW unit to draw this task. */
+        if(t->preferred_draw_unit_id != DRAW_UNIT_ID_VGLITE)
+            return LV_DRAW_UNIT_IDLE;
+    }
+    else {
+        /* Fake unsupported tasks as ready. */
+        if(t->preferred_draw_unit_id != DRAW_UNIT_ID_VGLITE) {
+            t->state = LV_DRAW_TASK_STATE_READY;
+
+            /* Request a new dispatching as it can get a new task. */
+            lv_draw_dispatch_request();
+
+            return 1;
+        }
+    }
```

Comment:
I'm still learning how draw dispatching works. For the `else` case (one unit), wouldn't this consume a SW unit and skip it being drawn by any renderer?

Reply:
Each unit dispatcher can have its own implementation.

In NXP VGLite case, before this else implementation, unsupported tasks were given to the SW draw unit.
But that means that LVGL should have had one main LVGL thread + VGLite draw uni thread + Sw draw unit thread.
We disabled the SW thread because it was kept just to cover "few" unsupported tasks. Those "few" unsupported tasks are not so common (like the box shadow). We can implement them in the future if really need it.

But the performance is much better when we have only one GPU draw unit to process everything because we do not have to wait for any task dependency and getting rid of one freertos thread (SW draw unit) also saves the CPU.